assets/js/modules-admin.js

```diff
@@ -202,36 +202,28 @@ jQuery( document ).ready( function () {
 		} );
 	}
 
-	// Get Course modules (if any) on course select change
-	var $courseSelect = jQuery( '#lesson-course-options' ),
-		$lessonModuleMetaboxSelectContainer = jQuery(
-			'div#lesson-module-metabox-select'
-		);
-
-	$courseSelect.on( 'change', function () {
-		if ( ! window.modulesAdmin.getCourseModulesNonce ) {
-			// console.log( 'missing modulesAdmin.getCourseModulesNonce' );
-			return;
-		}
-		var courseId = $courseSelect.val(),
-			data = {
-				security: window.modulesAdmin.getCourseModulesNonce,
-				action: 'sensei_get_course_modules',
+	// Refresh the modules meta box on course select change.
+	jQuery( '#lesson-course-options' ).on( 'change', function () {
+		const lessonId = jQuery( '#post_ID' ).val();
```

Comment:
> I think I may have missed this in the other PR, but how confident are you that #post_id will always be available in the editor?

The `#post_id` element is part of the `the_block_editor_meta_boxes` core function, so I've thought it's pretty safe to use, just because it's part of blocks core. I just found another way to get the ID using `wp.data.select( 'core/editor' ).getCurrentPostId()`. Should I go with the second option?

>Also, what do you think about doing a check that both lessonId and courseId have a value before making the AJAX call? We could save this call if either is not set for some reason.

The `courseId` can be empty, so we shouldn't check its value. This is the case when the user selects "None" as a course.

I'm not a big fan of code failing silently, so that's why I'm not checking the `lessonId`. It makes bugs very difficult to find and debug. If for some reason the `lessonId` is empty, there should be an error somewhere (in the backend in this case), because this is unexpected behavior. I'd love to hear your thoughts.

Reply:
> The #post_id element is part of the the_block_editor_meta_boxes core function, so I've thought it's pretty safe to use, just because it's part of blocks core. I just found another way to get the ID using wp.data.select( 'core/editor' ).getCurrentPostId(). Should I go with the second option?

Nope, if you think it's safe then that's good enough for me! 🙂 I usually see `wp_localize_script` used for this sort of thing, but this way might just be better.

> I'm not a big fan of code failing silently, so that's why I'm not checking the lessonId. It makes bugs very difficult to find and debug. If for some reason the lessonId is empty, there should be an error somewhere (in the backend in this case), because this is unexpected behavior.

Reply:
> Nope, if you think it's safe then that's good enough for me! 🙂 I usually see wp_localize_script used for this sort of thing, but this way might just be better.

Thinking about it now, maybe the safer option would be to use the `wp.data.select( 'core/editor' ).getCurrentPostId()` (the selector was created 5 years ago) just because it's a [documented API](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-editor/#getcurrentpostid). I much prefer that than relying on a DOM element. I've updated the code in https://github.com/Automattic/sensei/pull/4421/commits/c4bc587967ef289f50fc0f343bb9a25d8c94ff3b. Let me know if you have any concerns.

> That makes sense. Another alternative might be to surface the missing lesson ID by logging it in the console in the frontend.

I've updated the code to use the wp data module, so I guess an error will be thrown in the frontend if that doesn't work for some reason.

Reply:
Would you mind rebasing to pull in Renatho's `package.json` changes so I can rebuild more easily?

Reply:
Yep, done.